### PR TITLE
EditFamilies from Overview Page

### PR DESCRIPTION
**Pull request by SindreVatnaland**
Families can now be edited from the admin overview page

### CODE REVIEW

File: ClientApp/src/components/admin/common/RecipientTable.tsx [hunk 1]
```diff
@@ -72,13 +79,15 @@ const DatatableRecipient: React.FC<Props> = ({
...
     <Container>
-      {data.map((recipient) => (
-        <Accordion expanded={expanded===recipient}
-          onChange={handleChange(recipient)}
+      {data.map((recipient, index) => (
+        <Accordion
+          expanded={selected == index}
           key={recipient.rowKey}
           className={classes.accordionContainer}
+          onClick={() => { index != selected ? setSelected(index) : setSelected(-1)}}
```
Comment (Sommerli) [inline ClientApp/src/components/admin/common/RecipientTable.tsx:87]: Should onClick use handleChange? 

handleChange function is unused after this changes

File: ClientApp/src/components/admin/overview/OverviewMacro.tsx [hunk 2]
```diff
@@ -78,14 +83,28 @@ const OverviewMacro: React.FC<IOverviewMacro> = ({ accessToken, location }) => {
...
+              dinner: newRecipient.dinner,
+              dessert: newRecipient.dessert,
+              note: newRecipient.note,
+              contactFullName: newRecipient.contactFullName,
+              contactEmail: newRecipient.contactEmail,
+              contactPhoneNumber: newRecipient.contactPhoneNumber,
+              institution: newRecipient.institution,
+              referenceId: newRecipient.referenceId,
+              familyMembers: newRecipient.familyMembers as Types.PersonType[],
+            } as Types.RecipientType
```
Comment (Sommerli) [inline ClientApp/src/components/admin/overview/OverviewMacro.tsx:107]: Maybe this function could easily map the properties?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

The usage of two classes with the same name is confusing. Why are they seperated and should they be seperate? 

Might be an idea to remove usage of one in favour of the other, or create a subclass

Comment (SindreVatnaland) [reply ClientApp/src/components/admin/overview/OverviewMacro.tsx]: I were no sure if that function were working due to the recipient objects originates from different types, but in fact it does!

I got it working writing it like this:
```cs
Object.assign({} as Types.RecipientType, recipient);
```

File: ClientApp/src/components/admin/common/RecipientTable.tsx (hunk 1 above)
Comment (SindreVatnaland) [reply ClientApp/src/components/admin/common/RecipientTable.tsx]: Created a handleSelectedAccordion() function!

File: ClientApp/src/components/institution/FamilyDialog.tsx [hunk 3]
```diff
@@ -10,9 +10,11 @@
  import Recipient from "../admin/overview/Recipient";
  import ApiService from "../../common/functions/apiServiceClass";
  import { useEffect, useState, useCallback} from "react";
+ import * as Types from "../admin/suggestedConnections/Types";
```
[the remark this answers is not in the document]
Comment (SindreVatnaland) [reply ClientApp/src/components/institution/FamilyDialog.tsx]: Moved this to ClientApp\src\common\components\Types.tsx

File: GiEnJul/Controllers/AdminController.cs [hunk 4]
```diff
@@ -288,6 +288,33 @@ public async Task<ActionResult> DeleteRecipientAsync([FromBody] DeleteRecipientD
...
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+            await _recipientRepository.InsertOrReplaceAsync(recipientToUpdate);
+            foreach (var person in recipientToUpdate.FamilyMembers)
+            {
+                await _personRepository.InsertOrReplaceAsync(person);
+            }
```
[the remark this answers is not in the document]
Comment (SindreVatnaland) [reply GiEnJul/Controllers/AdminController.cs:314]: Like this?:
```cs
public async Task<ActionResult> PutRecipientAsync([FromBody] PutRecipientDto recipientDto)
        {
            var recipientUpdated = _mapper.Map<Recipient>(recipientDto);
            var recipientToUpdate = await _recipientRepository.GetRecipientAsync(recipientDto.PartitionKey, recipientDto.RowKey);

            try
            {
                foreach (var prop in recipientUpdated.GetType().GetProperties())
                {
                    var value = prop.GetValue(recipientUpdated);
                    if (value != null) prop.SetValue(recipientToUpdate, value);
                }


                await _recipientRepository.InsertOrReplaceAsync(recipientToUpdate);
                foreach (var person in recipientToUpdate.FamilyMembers)
                {
                    await _personRepository.InsertOrReplaceAsync(person);
                }
            }
            catch (Exception ex)
            {
                throw ex;
            }
            return Ok();
        }
```

File: ClientApp/src/components/institution/FamilyDialog.tsx [hunk 5]
```diff
@@ -70,7 +71,7 @@ const FamilyDialog: React.FC<IFamilyDialog> = ({ open, accessToken,institution,
                      <DialogContentText id="alert-dialog-description">
                          <p>Liste over familier</p>
                      </DialogContentText>
-                     <Recipient data={recipientData} handleRecipientChange={handleRecipientChange}/>
+                     <Recipient data={recipientData} accessToken={""} refreshRecipients={() => ""}/>
```
Comment (Sommerli) [inline ClientApp/src/components/institution/FamilyDialog.tsx:74]: Should this include accessToken if institutions should want to change families?

**Review by Sommerli — APPROVED** [review]

**Pull request merged**